src/ukify/ukify.py

```diff
@@ -918,6 +935,117 @@ def make_uki(opts):
 
     print(f"Wrote {'signed' if sign_args_present else 'unsigned'} {opts.output}")
 
+def sign_uki(opts):
+    """
+    Unpack UKI, replace cmdline, measure and sign PCR, then repack and sign it.
+    We have to repack as PCR signature should be before the linux section.
+    """
+    if not opts.sb_key and not opts.sb_cert_name:
+        raise ValueError("Signing arguments required to sign UKI")
+
+    pe = pefile.PE(opts.uki, fast_load=True)
+
+    uki = UKI(opts.uki)
+    linux_section = None
+    sign_kernel = opts.sign_kernel
+
+    measured_sections = [
+        '.osrel',
+        '.cmdline',
+        '.dtb',
+        '.uname',
+        '.splash',
+        '.pcrpkey',
+        '.initrd',
+        '.ucode',
+        '.linux'
+    ]
+    additional_sections = []
+    for section in pe.sections:
+        name = section.Name.rstrip(b"\x00").decode()
+
+        if name not in measured_sections:
+            continue
+
+        if name == ".cmdline" and opts.cmdline:
+            cmdline = opts.cmdline
+            if opts.append_cmdline:
+                cmdline = section.get_data().rstrip(b"\x00").decode() + " " + cmdline
+            uki_section = Section.create(name, cmdline, measure=True)
+            additional_sections += [uki_section]
+        else:
+            uki_section = Section.create(name, section.get_data(), measure=True)
+
+        if name == ".linux":
+            linux_section = uki_section
+        else:
+            uki.add_section(uki_section)
+
+    if linux_section is None:
+        raise ValueError("No linux section found! Is this a valid UKI?")
+
+    if opts.signtool == 'sbsign':
+        sign_tool = find_sbsign(opts=opts)
+        sign = sbsign_sign
+        verify_tool = SBVERIFY
+    else:
+        sign_tool = find_pesign(opts=opts)
+        sign = pesign_sign
+        verify_tool = PESIGCHECK
+
+    if sign_kernel is None:
+        # figure out if we should sign the kernel
+        sign_kernel = verify(verify_tool, opts, linux=linux_section.content)
+
+    if sign_kernel:
+        linux_signed = tempfile.NamedTemporaryFile(prefix='linux-signed')
```

Comment:
The name is not new, but maybe it would make sense to change it slightly and add a `ukify-` prefix to make it more easily identifiable in case the temporary file persists, but that's something @keszybz should decide.

Also a personal nit the name here would be `linux-signed12346`, i.e. the random string gets directly appended, having looked at a lot of temporary files, it's nicer if there is a further character separating this like a dot or a dash.

Both points hold for the other two tempfiles in this function as well.

Reply:
@behrmann Changed, but there are some more occurences througout the file. I could change these as well?